File: 1944-number-of-visible-people-in-a-queue/1944-number-of-visible-people-in-a-queue.js
/**
 * @param {number[]} heights
 * @return {number[]}
 */
var canSeePersonsCount = function(heights) {
    
    
    const res = []
    const stack = []
    for(let i = heights.length - 1; i >=0; i--){
        let curr = heights[i]
        let size = 0
        
        while(stack.length && stack[stack.length - 1] <= curr){
            size++
            stack.pop()
        }
        
        if(stack.length)
            size++
         stack.push(curr)
        res.push(size)
    }
    
    return res.reverse()
    
};

File: 1944-number-of-visible-people-in-a-queue/1944-number-of-visible-people-in-a-queue.py
class Solution:
    def canSeePersonsCount(self, heights: List[int]) -> List[int]:
        
        
        
        stack = []
        res = []
        
        for curr in heights[::-1]:
            size = 0
            while stack and stack[-1] <= curr :
                size += 1
                stack.pop()
            if stack :
                size+=1
            res.append(size)
            stack.append(curr)
        return res[::-1]     
        